src/dra/843B0.c

```diff
@@ -414,8 +414,126 @@ void EntityHolyWater(Entity* entity) {
 INCLUDE_ASM("dra/nonmatchings/843B0", EntityHolyWaterBreakGlass);
 
 // green flame when holy water explodes
-INCLUDE_ASM("dra/nonmatchings/843B0", EntityHolyWaterFlame);
+void EntityHolyWaterFlame(Entity* self) {
+    s16 sp10[5];
+    s16 sp20[5];
+    s16 pad[2];
+    Primitive* prim;
+    s16 temp_v0_4;
+    s16 angleTemp;
+    s32 angle;
+    s32 temp_v0;
+    s32 i;
+    s32 hex80;
+    u8 randR;
+    u8 randG;
+    u8 randB;
+    u8 primUBase;
+    u8 primVBase;
+
+    s16* primYPtr;
 
+    primUBase = D_800B0688[(g_GameTimer & 7)].x;
+    primVBase = D_800B0688[(g_GameTimer & 7)].y;
+    switch (self->step) {
+    case 0:
+        randR = (rand() & 0x1F) | 0x40;
+        randG = (rand() & 0x1F) | 0x80;
+        randB = (rand() & 0x1F) | 0x60;
+        self->primIndex = AllocPrimitives(4U, 4);
```

Comment:
```suggestion
        self->primIndex = AllocPrimitives(PRIM_GT4, 4);
```

Reply:
Ah, of course. Good catch! Fixing now. Thanks!